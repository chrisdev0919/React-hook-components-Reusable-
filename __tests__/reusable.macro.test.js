const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");
const prettier = require("prettier");

const noUsageCode = `import { reuseState } from "../src/reuseable.macro";`;

const reuseStateCode = `
  import { reuseState } from "../src/reuseable.macro";
  
  const counter = () => {
    const [count, setCount] = reuseState(0);
  }
`;

const shorthandSyntaxCode = `
  import { reuseState } from "../src/reuseable.macro";
  
  const counter = () => reuseState(0)
`;

const reuseRefCode = `
  import { reuseRef } from "../src/reuseable.macro";
  
  const counter = () => {
    const count = reuseRef(0);
  }
`;

const reuseCallbackWithDependenciesCode = `
  import { reuseCallback } from "../src/reuseable.macro";
  
  const counter = () => {
    const count = reuseCallback(() => {}, []);
  }
`;

const reuseCallbackNoDependenciesCode = `
  import { reuseCallback } from "../src/reuseable.macro";
  
  const counter = () => {
    const count = reuseCallback(() => {});
  }
`;

const reuseReducerWithInitialState = `
  import { reuseReducer } from "../src/reuseable.macro";
  
  const reducer = (state, action) => state;
  const initialState = {};
  
  const counter = () => {
    const count = reuseReducer(reducer, initialState);
  }
`;

const reuseReducerCode = `
  import { reuseReducer } from "../src/reuseable.macro";
  
  const reducer = (state, action) => state;
  
  const counter = () => {
    const count = reuseReducer(reducer);
  }
`;

const reuseMemoCode = `
  import { reuseMemo } from "../src/reuseable.macro";
  
  const counter = () => {
    reuseMemo(() => {
    
    }, []);
  }
`;

const reuseMemoWithVariableCode = `
  import { reuseMemo } from "../src/reuseable.macro";
  
  const counter = () => {
    const count = reuseMemo(() => null)
  }
`;

const reuseEffectCode = `
  import { reuseEffect } from "../src/reuseable.macro";
  
  const counter = () => {
    reuseEffect(() => {
    
    }, []);
  }
`;

const InvalidDefaultImportCode = `
import customName from '../src/reuseable.macro';

const counter = () => {
  const [count, setCount] = customName.dodoState(0);
}
`;

const InvalidImportCode = `
import { dodoState } from "../src/reuseable.macro";

const counter = () => {
  const [count, setCount] = dodoState(0);
}
`;

pluginTester({
  title: 'macro',
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5", parser: 'babel' });
  },
  tests: {
    "no usage": noUsageCode,
    "should work with { reuseState }": reuseStateCode,
    "should work with { reuseState } shorthand": shorthandSyntaxCode,
    "should work with { reuseRef }": reuseRefCode,
    "should work with { reuseCallback } with dependencies": reuseCallbackWithDependenciesCode,
    "should work with { reuseCallback } without dependencies": reuseCallbackNoDependenciesCode,
    "should work with { reuseReducer } with initial state": reuseReducerWithInitialState,
    "should work with { reuseReducer } without initial state": reuseReducerCode,
    "should work with { reuseMemo }": reuseMemoCode,
    "should work with { reuseMemo } with variable": reuseMemoWithVariableCode,
    "should work with { reuseEffect }": reuseEffectCode,
    'should work with custom import name': {
      code: InvalidDefaultImportCode,
      error: true,
      snapshot: false
    },
    "invalid import": {
      code: InvalidImportCode,
      error: true,
      snapshot: false
    },
  },
});
