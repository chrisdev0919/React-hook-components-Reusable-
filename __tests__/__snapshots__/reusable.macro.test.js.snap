// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macro no usage: no usage 1`] = `
"
import { reuseState } from \\"../src/reuseable.macro\\";

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseState as _reuseState } from \\"reusable\\";

"
`;

exports[`macro should work with { reuseCallback } with dependencies: should work with { reuseCallback } with dependencies 1`] = `
"
import { reuseCallback } from \\"../src/reuseable.macro\\";

const counter = () => {
  const count = reuseCallback(() => {}, []);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseCallback as _reuseCallback } from \\"reusable\\";

const counter = () => {
  const count = _reuseCallback(() => {}, [], \\"count\\");
};

"
`;

exports[`macro should work with { reuseCallback } without dependencies: should work with { reuseCallback } without dependencies 1`] = `
"
import { reuseCallback } from \\"../src/reuseable.macro\\";

const counter = () => {
  const count = reuseCallback(() => {});
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseCallback as _reuseCallback } from \\"reusable\\";

const counter = () => {
  const count = _reuseCallback(() => {}, undefined, \\"count\\");
};

"
`;

exports[`macro should work with { reuseEffect }: should work with { reuseEffect } 1`] = `
"
import { reuseEffect } from \\"../src/reuseable.macro\\";

const counter = () => {
  reuseEffect(() => {
  
  }, []);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseEffect as _reuseEffect } from \\"reusable\\";

const counter = () => {
  _reuseEffect(() => {}, [], \\"counter reuseEffect (0)\\");
};

"
`;

exports[`macro should work with { reuseMemo } with variable: should work with { reuseMemo } with variable 1`] = `
"
import { reuseMemo } from \\"../src/reuseable.macro\\";

const counter = () => {
  const count = reuseMemo(() => null)
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseMemo as _reuseMemo } from \\"reusable\\";

const counter = () => {
  const count = _reuseMemo(() => null, undefined, \\"count\\");
};

"
`;

exports[`macro should work with { reuseMemo }: should work with { reuseMemo } 1`] = `
"
import { reuseMemo } from \\"../src/reuseable.macro\\";

const counter = () => {
  reuseMemo(() => {
  
  }, []);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseMemo as _reuseMemo } from \\"reusable\\";

const counter = () => {
  _reuseMemo(() => {}, [], \\"counter reuseMemo (0)\\");
};

"
`;

exports[`macro should work with { reuseReducer } with initial state: should work with { reuseReducer } with initial state 1`] = `
"
import { reuseReducer } from \\"../src/reuseable.macro\\";

const reducer = (state, action) => state;
const initialState = {};

const counter = () => {
  const count = reuseReducer(reducer, initialState);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseReducer as _reuseReducer } from \\"reusable\\";

const reducer = (state, action) => state;

const initialState = {};

const counter = () => {
  const count = _reuseReducer(reducer, initialState, \\"count\\");
};

"
`;

exports[`macro should work with { reuseReducer } without initial state: should work with { reuseReducer } without initial state 1`] = `
"
import { reuseReducer } from \\"../src/reuseable.macro\\";

const reducer = (state, action) => state;

const counter = () => {
  const count = reuseReducer(reducer);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseReducer as _reuseReducer } from \\"reusable\\";

const reducer = (state, action) => state;

const counter = () => {
  const count = _reuseReducer(reducer, undefined, \\"count\\");
};

"
`;

exports[`macro should work with { reuseRef }: should work with { reuseRef } 1`] = `
"
import { reuseRef } from \\"../src/reuseable.macro\\";

const counter = () => {
  const count = reuseRef(0);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseRef as _reuseRef } from \\"reusable\\";

const counter = () => {
  const count = _reuseRef(0, \\"count\\");
};

"
`;

exports[`macro should work with { reuseState } shorthand: should work with { reuseState } shorthand 1`] = `
"
import { reuseState } from \\"../src/reuseable.macro\\";

const counter = () => reuseState(0)

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseState as _reuseState } from \\"reusable\\";

const counter = () => _reuseState(0, \\"counter\\");

"
`;

exports[`macro should work with { reuseState }: should work with { reuseState } 1`] = `
"
import { reuseState } from \\"../src/reuseable.macro\\";

const counter = () => {
  const [count, setCount] = reuseState(0);
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { reuseState as _reuseState } from \\"reusable\\";

const counter = () => {
  const [count, setCount] = _reuseState(0, \\"count\\");
};

"
`;
